lasy/laser.py

```diff
@@ -160,6 +160,46 @@ def normalize(self, value, kind="energy"):
         else:
             raise ValueError(f'kind "{kind}" not recognized')
 
+    def apply_optics(self, optical_element):
+        """
+        Propagate the laser pulse through a thin optical element.
+
+        Parameter
+        ---------
+        optical_element: an :class:`.OpticalElement` object (optional)
+            Represents a thin optical element, through which the laser
+            propagates.
+        """
+        # Transform the field from temporal to frequency domain
+        time_axis_indx = -1
+        field_fft = np.fft.ifft(self.grid.field, axis=time_axis_indx, norm="backward")
```

Comment:
The implementation is not efficient currently, as it performs FFTs forward and backward.
This will be fixed in a future PR, by keeping the Fourier-space fields in memory and having a flag that keeps track of this.